overlord/hookstate/hookmgr.go

```diff
@@ -288,6 +310,15 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 		return nil, err
 	}
 
+	// add timeout handling
```

Comment:
I suppose we would miss the Stop but given we don't run lots of these all the time, it's probably a non issue